CONTRIBUTING.md

```diff
@@ -4,63 +4,110 @@ Thank you for your interest in contributing to the CircleCI documentation.
+This will take you to the source file on GitHub,
+where you can submit a pull request for your changes.
 
-For larger edits or new articles, you'll want to [set up a local environment](README-local-development.md) for editing or adding articles. When you are finished with your changes, create a pull request from your branch by following [GitHub's guide](https://help.github.com/articles/creating-a-pull-request-from-a-fork/).
+For larger edits or new documents,
+[set up a local environment](README-local-development.md).
+When you are satisfied with your changes,
+create a pull request from your branch
+by following [GitHub's guide](https://help.github.com/articles/creating-a-pull-request-from-a-fork/).
 
-In order to help those watching for new modifications and additions to the documentation, it is suggested that your pull request title be descriptive enough to help those watching the repository get a general idea of what is being changed and why. Adding a description with more information&mdash;if necessary&mdash;is a bonus!
+### Titles and Descriptions
+
+Pull request titles should be descriptive enough
+for reviewers to understand *what* is being changed.
+Some ways of doing this are better than others:
 
 | Original Pull Request Title | Better Title                                                               |
 |-----------------------------|----------------------------------------------------------------------------|
 | _Updating file.md_          | _Indicate support for environment variables in context paths_            |
 | _Sidebar changes_           | _Move Deployment to its own navigation section for better organization_  |
 
+Every pull request should have a description
+that explains *why* the change is being made.
+The description adds context
+that is critical for reviewers when giving feedback.
+
 For more tips, see GitHub's blog entry on [how to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request).
 
 # Style Guide
 
-For basic technical writing style, see [Technical writing style](https://en.wikiversity.org/wiki/Technical_writing_style).
-
-We also have a short list of guidelines that take precedence over the general guide above. We don't have a lot of rules, but we do try to be consistent with the ones we have.
+We do not have a lot of rules,
+but we do consistently apply them.
 
 ## Markdown
 
-To keep the emphasis on prose, use [markdown](http://commonmark.org/help/) over plain HTML.
+To keep the emphasis on prose,
+use markdown instead of plain HTML.
+Refer to the [Markdown documentation](https://daringfireball.net/projects/markdown/syntax) for more details.
+
+## Links
+
+### Explicit Names
+
+Link text should explicitly name the document or section
+to which it is linking.
+
+For example,
+if you reference a section called "Adding Environment Variables in the App",
+write, "Refer to the 'Adding Environment Variables in the App' section of the Environment Variables document."
```

Comment:
might as well show how to make the markdown relative link here.

Reply:
These are GitHub-specific, right? Before we show _how_, I'd rather know _when_ and _why_ we'd use relative links at all.

Reply:
We always use them for internal links in case we decide to change where the `/docs1 directory lives at some point (and probably for other reasons) and we do it like this in Markdown:
`[Adding Environment Variables]({{ site.baseurl }}/2.0/env-vars/#adding-environment-variables)`

Reply:
Sorry, I'm an idiot. I was thinking of something completely different when you said relative link. Fully agree with this.